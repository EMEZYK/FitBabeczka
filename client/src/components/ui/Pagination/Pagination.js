import React from "react";
import { PaginationList, PageNumber } from "./Pagination.styled";
import { useState } from "react";
import { PaginatedList } from "../../views/AllRecipesPage/AllRecipes.styled";
import { NextPrevButton } from "../../views/AllRecipesPage/AllRecipes.styled";

export const Pagination = ({
  totalRecipes,
  recipesPerPage,
  paginate,
  thisPage,
  numberOfPages,
  setPageNumber,
  pageNumber,
}) => {
  const pageNumbers = new Array(numberOfPages).fill(null).map((v, i) => i);

  const [activePageNumber, setActivePageNumber] = useState(0);

  const toggleActive = (pageNum) => {
    setActivePageNumber(pageNum);
  };

  const nextPage = () => {
    setPageNumber(Math.min(numberOfPages - 1, pageNumber + 1));
  };

  const previousPage = () => {
    setPageNumber(Math.max(0, pageNumber - 1));
  };

  return (
    <PaginatedList>
      <NextPrevButton
        onClick={() => {
          previousPage();
          toggleActive(activePageNumber - 1);
        }}
      >
        <a href="#">&laquo;</a>
      </NextPrevButton>
      <PaginationList>
        {pageNumbers.map((page) => (
          <PageNumber
            className={activePageNumber === page ? "active" : ""}
            key={page}
            onClick={() => {
              toggleActive(page);
              paginate(page);
            }}
          >
            <a href="#/" className={activePageNumber === page ? "active" : ""}>
              {page + 1}
            </a>
          </PageNumber>
        ))}
      </PaginationList>
      <NextPrevButton
        onClick={() => {
          toggleActive(activePageNumber + 1);
          nextPage();
        }}
      >
        <a href="#" onClick={nextPage}>
          &raquo;
        </a>
      </NextPrevButton>
    </PaginatedList>
  );
};
