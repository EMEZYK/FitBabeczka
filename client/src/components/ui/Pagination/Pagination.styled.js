import styled from "styled-components";

export const PaginationList = styled.ul`
  list-style: none;
`;
