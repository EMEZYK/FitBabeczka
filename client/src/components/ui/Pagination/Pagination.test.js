import React from "react";
import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom";
import { Pagination } from "./Pagination";
import { PageNumber } from "./Pagination.styled";
import * as renderer from "react-test-renderer";
import "jest-styled-components";

describe("Pagination components", () => {
  afterEach(cleanup);

  test("Check if takes a snapshot',", () => {
    const { asFragment } = render(<Pagination />);
    expect(asFragment(<Pagination />)).toMatchSnapshot();
  });

  test("Check if page number change bgc when item is active", () => {
    const pagination = renderer.create(<PageNumber />).toJSON();
    expect(pagination).toHaveStyleRule("background-color", "#cc9933", {
      modifier: " &.active ",
    });
  });

  test("Check if page number change color when item is active", () => {
    const pagination = renderer.create(<PageNumber />).toJSON();
    expect(pagination).toHaveStyleRule("color", "#fff", {
      modifier: " &.active ",
    });
  });
});
