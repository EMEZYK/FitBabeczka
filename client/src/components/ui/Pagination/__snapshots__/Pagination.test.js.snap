// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination components Check if takes a snapshot', 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0 0 0 0.5rem;
}

.c3 {
  width: 2rem;
  justify-items: center;
  text-align: center;
  padding: 0.3rem;
  margin: 0 0.5rem 0 0;
  background-color: #fff;
  border: 1px solid #aaa;
  cursor: pointer;
}

.c3.active {
  background-color: #cc9933;
  color: #fff;
}

.c3 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

.c3 a.active {
  color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 1rem 0;
}

.c1 {
  width: 2rem;
  padding: 0.3rem;
  cursor: pointer;
  background-color: #fff;
  border: 1px solid #ddd;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
}

<div
    class="c0"
    data-testid="pagination"
  >
    <button
      class="c1"
    >
      <a
        href="#"
      >
        «
      </a>
    </button>
    <ul
      class="c2"
    >
      <li
        class="c3 active"
        data-testid="page-number"
      >
        <a
          class="active"
          href="#/"
        >
          1
        </a>
      </li>
    </ul>
    <button
      class="c1"
    >
      <a
        href="#"
      >
        »
      </a>
    </button>
  </div>
</DocumentFragment>
`;
